packages/eslint-plugin/src/rules/no-extra-parens.ts

```diff
@@ -0,0 +1,33 @@
+import { TSESTree, AST_NODE_TYPES } from '@typescript-eslint/typescript-estree';
+import baseRule from 'eslint/lib/rules/no-extra-parens';
+import * as util from '../util';
+
+type Options = util.InferOptionsTypeFromRule<typeof baseRule>;
+type MessageIds = util.InferMessageIdsTypeFromRule<typeof baseRule>;
+
+export default util.createRule<Options, MessageIds>({
+  name: 'no-extra-parens',
+  meta: {
+    type: 'layout',
+    docs: {
+      description: 'disallow unnecessary parentheses',
+      category: 'Possible Errors',
+      recommended: false,
+    },
+    fixable: 'code',
+    schema: baseRule.meta.schema,
+    messages: baseRule.meta.messages,
+  },
+  defaultOptions: ['all'],
+  create(context) {
+    const rules = baseRule.create(context);
+
+    return Object.assign({}, rules, {
+      MemberExpression(node: TSESTree.MemberExpression) {
+        if ((node.object as any).type !== AST_NODE_TYPES.TSAsExpression) {
```

Comment:
Instead of doing a cast here - you should fix the type for MemberExpression.

In particular, you should update the type of `LeftHandSideExpression` to include `TSAsExpression`

Reply:
Hi @bradzacher , I agree that it's a "quick and dirty" solution :)

I've tried to update the type of `LeftHandSideExpression` to 
```typescript
// ts-estree.ts
export type LeftHandSideExpression =
  | CallExpression
  | ClassExpression
  | ClassDeclaration
  | FunctionExpression
  | LiteralExpression
  | MemberExpression
  | PrimaryExpression
  | TaggedTemplateExpression
  | TSNonNullExpression
  | TSAsExpression;
```


but I've stuck with an error 
`Error:(27, 13) TS2367: This condition will always return 'true' since the types 'AST_NODE_TYPES.ArrayExpression | AST_NODE_TYPES.ArrayPattern | AST_NODE_TYPES.BigIntLiteral | AST_NODE_TYPES.CallExpression | AST_NODE_TYPES.ClassDeclaration | AST_NODE_TYPES.ClassExpression | AST_NODE_TYPES.FunctionExpression | AST_NODE_TYPES.Identifier | AST_NODE_TYPES.Import | AST_NODE_TYPES.JSXElement | ... 12 m...' and 'AST_NODE_TYPES.TSAsExpression' have no overlap.` in 

```typescript
if (node.object.type !== AST_NODE_TYPES.TSAsExpression) {
    return rules.MemberExpression(node);
}
```

I will check more why it happens

Reply:
*UDP:* It appeared that it was a problem in TS Language Service in WebStorm (maybe I should switch to VSCode, hehe). I've restarted the Language Service and error has gone.
I will update this PR.